### ISSUE TITLE
possible compositor issues with latest brewmaster and Intel Graphics (Coffee Lake)

### ISSUE THREAD

**Issue by oliverbr** [ValveSoftware/SteamOS#676]
#### Your system information

* **Steam client version:** 1556574584 (built Apr 29 2019, at 20:58:23)
* **SteamOS version:** SteamOS 2.0 update 1:2.189
* Opted into Steam client beta?: No
* **Opted into SteamOS beta?:** No
* **Have you checked for updates in Settings > System?:** Yes (updated today, 2019-05-06 19:30 CEST)

* **Platform:** Intel NUC8i7BEH
* **Video Card:** reported as "Intel Open Source Technology Center Mesa DRI Intel(R) HD Graphics (Coffeelake 3x8 GT3) x86/MMX/SSE2"
* **Driver version:** 3.0 Mesa 18.3.4

#### Please describe your issue in as much detail as possible:

After updating to the latest version, XCOM 1 and Rogue Legacy no longer appear in the foreground and are instead executed "behind" SteamOS Big Picture. Sound is played and the games are controllable. The cursor is changed to the XCOM-specific cursor when the game starts.

When opening the SteamOS overlay, the same behavior can be observed.

The error is present in a clean install and is not remediated by installing the missing i915 firmware files from the Linux Kernel repository.

#### Steps for reproducing this issue:

Update from previous SteamOS stable release to 2.189

The only interesting log data I could find after looking around in the various log locations is the `steam` user's `.xsession-errors` file, which is available at http://sprunge.us/gWkckx
The SteamOS logs as described by the wiki do not seem relevant, although I can add them, if necessary.

If there's anything more I can add, please let me know.

**Comment by dubigrasu** [comment]
There is a fork of steamcompmgr that is supposed to fix this behavior, you can use it for testing, see if it solves the issue.
https://github.com/alkazar/steamos-compositor

**Comment by kisak-valve** [comment]
Possibly related to #612.

**Comment by oliverbr** [comment]
Thanks for pointing me towards the compositor fork! I cloned the current master of the repository and installed the replacement compositor via the bundled script.

This did indeed solve part of the issue, since the overlay is visible again - I can now exit said games without having to blindly navigate the overlay (or, more often than not, reboot ;) )

However, the issues with the games not displaying remains. 
Rouge Legacy now blanks the screen but does not do much else; XCOM gives me the selection between using the Add-on Enemy Within or not, but does not render anything beyond that. This initial section is not rendered in the game's engine, which is probably the reason it can be shown.

I will try to fiddle around with some things I am comfortable with modifying. If there are any more hints towards debugging this behaviour, I'm all ears!

**Comment by alkazar** [comment]
@oliverbr 

I am the developer on the compositor fork. I just tested Rogue Legacy (limited internet so can't download XCOM) and it worked fine for me on both the original compositor and my fork. I am also using intel graphics but am using the compositor on Arch Linux, not SteamOS. Perhaps the issue is not compositor related?

Do other games work? Are there any other games that have issues?

**Comment by alkazar** [comment]
@oliverbr 

I am the developer on the compositor fork. I just tested Rogue Legacy (limited internet so can't download XCOM) and it worked fine for me on both the original compositor and my fork. I am also using intel graphics but am using the compositor on Arch Linux, not SteamOS. The original compositor has not been updated since August 2018. Perhaps the issue is not compositor related?

Do other games work? Are there any other games that have issues?

**Comment by oliverbr** [comment]
@alkazar 

> I am also using intel graphics but am using the compositor on Arch Linux, not SteamOS. 

I'm no expert here, but that would probably mean that you have a different Mesa version and Kernel configuration differences. 
Attempting to switch to something non-SteamOS is something I will try soon.

>The original compositor has not been updated since August 2018. Perhaps the issue is not compositor related?

It does seem more likely now, even if installing your fork did at least make the overlay visible again. The compositor was my first guess, given the focussing issues I described above.

> Do other games work? Are there any other games that have issues?

I will try to test some other games today, but from what I've seen so far, my guess would be that they will exhibit similar behaviours.

**Comment by dubigrasu** [comment]
FWIW, I tried XCOM (I don't have Rogue Legacy) on SteamOS actual, and I see no issues.

**Comment by oliverbr** [comment]
So I tried The Beginner's Guide and VVVVV this time around.

|game|vanilla compositor|patched compositor|
|:-|-|-|
|The Beginner's Guide|slanted back rectangle in top left part of the screen, Steam still shown underneath|black screen|
|VVVVV|playable, but with two columns of horizontal bars drawn over the game|untested|
|SteamOS overlay|slanted back rectangle in top left corner of the screen|shows|

Since this isn't really getting me anywhere, I'll try Arch next to rule out hardware issues. (My NUC is new, but you never know...)

**Comment by alkazar** [comment]
@oliverbr yeah, that doesn't sound normal. Something is definitely wrong. Trying a different distro seems like a good next step.
